app/models/jobstatusdata.rb

```diff
@@ -121,13 +123,27 @@ def extended_data_default(info)
     self.output_path = ''
 
     output_pathname = Pathname.new(ENV["HOME"])
-    self.terminal_path = OodAppkit.shell.url(path: (output_pathname.writable? ? output_pathname : ENV["HOME"]))
-    self.fs_path = OodAppkit.files.url(path: (output_pathname.writable? ? output_pathname : ENV["HOME"]))
+    self.file_explorer_url = { path: output_pathname }
+    self.shell_url = { path: output_pathname, cluster: self.cluster }
+
     self
   end
 
   private
 
+    def file_explorer_url=(path: path())
+      path = (path.writable? ? path : ENV["HOME"]).to_s
+
+      @file_explorer_url = OodAppkit.files.url(path: path).to_s
+    end
+
+    def shell_url=(path: path(), cluster: cluster())
+      path = (path.writable? ? path : ENV["HOME"]).to_s
+      host = OODClusters[cluster].login.host if OODClusters[cluster].login_allow?
+
+      @shell_url = host ? OodAppkit.shell.url(path: path, host: host).to_s : OodAppkit.shell.url(path: path).to_s
```

Comment:
My original solution was very similar to this but I was instructed to go the current route instead at https://github.com/OSC/ood-activejobs/pull/114#issuecomment-296253397